### PR TITLE
Allow linking directly to specific campaign updates.

### PR DESCRIPTION
**Pull request by DFurnes**
### What does this PR do?
This pull request refactors how blocks are rendered & adds a `/blocks/:id` page for block permalinks.

📦 I extracted the logic to determine which block to use into the `Block` component (which just accepts a Contentful JSON blob) so we can more easily display blocks outside of the feed. This made it much easier to display a block outside of the feed, and makes things more readable to boot.

⁉️ I tried out putting a handful of block related components in the `/components/Block` directory (including the container), and then just exporting a "table of contents" from the index (following along with some of the other recent organizational ideas.) Give it a [peek](https://github.com/DoSomething/phoenix-next/tree/cb20eeeef12c5c1524f8f744d9a7df118fc98376/resources/assets/components/Block).

🔗 And finally, I added a `/blocks/:id` route and little permalinks (click the "Campaign Update" title to get the link for any update) so that we can easily link to any specific campaign updates. I'm pushing this up now to get a MVP shipped, but the plan is to have any block be linkable & to display 'em in a modal. Here's an example: [`/us/campaigns/sincerely-us/blocks/1SLj7iLgp6QWs4Ma2USWMm`](https://phoenix-next-stage-pr-310.herokuapp.com/us/campaigns/sincerely-us/blocks/1SLj7iLgp6QWs4Ma2USWMm).

### Any background context you want to provide?
🍃

### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

**Comment by deadlybutter** [comment]
Will review the full thing after the shipment form stuff, but love the file organization for the block component here.

**Comment by DFurnes** [comment]
[(5) As a messaging team member, I want to be able to link to specific blocks in the campaign feed.](https://trello.com/c/B1nhU2Gw/634-5-as-a-messaging-team-member-i-want-to-be-able-to-link-to-specific-blocks-in-the-campaign-feed)

File: resources/assets/components/Block/block-wrapper.scss [hunk 1]
```diff
@@ -15,11 +15,20 @@
...
   margin: (- $half-spacing) (- $half-spacing) $base-spacing;
   padding: $half-spacing;
 }
+
+.block-wrapper__title a {
+  color: $black;
+}
+
+.block-wrapper__title a:hover {
+  color: rgba($black, 0.75);
```
Comment (deadlybutter) [inline resources/assets/components/Block/block-wrapper.scss]: I had no idea you could put a variable in this 🤓

File: resources/assets/components/Feed/index.js [hunk 2]
```diff
@@ -1,37 +1,23 @@
...
  * @returns {XML}
  */
-const renderFeedItem = (block, index) => {
-  const BlockComponent = get({
-    'campaign_update': CampaignUpdateBlock,  // eslint-disable-line quote-props
-    'join_cta': CallToActionContainer, // eslint-disable-line quote-props
-    'reportbacks': ReportbackBlock, // eslint-disable-line quote-props
-    'static': StaticBlock, // eslint-disable-line quote-props
-  }, block.fields.type, PlaceholderBlock);
+import PropTypes from 'prop-types';
```
Comment (deadlybutter) [inline resources/assets/components/Feed/index.js]: I'm not sure I understand why we moved these imports below the comment for the `renderFeedItem`?

File: resources/assets/components/Block/Block.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      return <StaticBlock fields={json.fields} />;
+
+    default:
+      return <PlaceholderBlock />;
+  }
+};
+
+Block.propTypes = {
+  // @TODO: Validate the shape of this JSON object with prop types.
+  json: PropTypes.object, // eslint-disable-line react/forbid-prop-types
```
Comment (weerd) [inline resources/assets/components/Block/Block.js]: eslint-disable strikes again! lol

File: resources/assets/components/Block/BlockWrapper.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  id: PropTypes.string,
+};
+
+BlockTitle.defaultProps = {
+  title: null,
+  id: null,
+};
+
+const BlockWrapper = props => (
+  <div className={classNames('block-wrapper', props.className)}>
```
Comment (weerd) [inline resources/assets/components/Block/BlockWrapper.js]: Not a merge blocker, but I could see this `div` being an `article` element. Especially now that they can be stand alone elements we link to. Just something to consider so we can add further semantics to the markup. I feel React makes it easy to just make all things `divs` 😬

File: resources/assets/components/Block/index.js [hunk 5]
```diff
@@ -1,29 +1,11 @@
-import PropTypes from 'prop-types';
-import React from 'react';
-import classNames from 'classnames';
-import './block.scss';
+// The <BlockWrapper> can be used to give the "block" visual
```
Comment (weerd) [inline resources/assets/components/Block/index.js]: I like this "table of contents" idea 😄

File: resources/assets/components/Feed/index.js (hunk 2 above)
Comment (weerd) [inline resources/assets/components/Feed/index.js]: I'm guessing maybe a PHPStorm thing that it did automatically? Not sure.

**Comment by weerd** [comment]
@DFurnes found a weird [visual bug](https://d2ppvlu71ri8gs.cloudfront.net/items/0B411b3v2M0X183d022g/Screen%20Recording%202017-06-15%20at%2012.05%20PM.gif?v=4b4d7046) to address on the linked block page.

File: resources/assets/components/Block/BlockWrapper.js (hunk 4 above)
Comment (DFurnes) [inline resources/assets/components/Block/BlockWrapper.js]: Yeah totally! 👍

File: resources/assets/components/Block/Block.js (hunk 3 above)
Comment (DFurnes) [inline resources/assets/components/Block/Block.js]: 😆

File: resources/assets/components/Feed/index.js (hunk 2 above)
Comment (DFurnes) [inline resources/assets/components/Feed/index.js]: Ugh, yeah. That' happened to me before. ⛈

Comment (DFurnes) [inline resources/assets/components/Feed/index.js]: Fixed in 66bbbba487b303475cb8cbcb253535837a522151.

File: resources/assets/components/Block/BlockWrapper.js (hunk 4 above)
Comment (DFurnes) [inline resources/assets/components/Block/BlockWrapper.js]: Changed in 4e28ab1.

File: resources/assets/components/Block/BlockWrapper.js [hunk 6]
```diff
@@ -21,10 +21,10 @@ BlockTitle.defaultProps = {
 };
 
 const BlockWrapper = props => (
-  <div className={classNames('block-wrapper', props.className)}>
+  <article className={classnames('block-wrapper', props.className)}>
     { props.title ? <BlockTitle title={props.title} id={props.id} /> : null }
     {props.children}
-  </div>
+  </article>
```
Comment (weerd) [inline resources/assets/components/Block/BlockWrapper.js]: 💃

**Comment by DFurnes** [comment]
@weerd Fixed in 4c15057!

**Pull request merged**